backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/service/ExpenseCalculatorService.java

```diff
@@ -263,11 +264,25 @@ private void persistMeta(List<Bill> bills,Map<String, String> metaInfo) {
     }
 
     public List<BillMapper> search(CalculatorSearchRequest calculatorSearchRequest) {
+        log.info("Validate calculatorSearchRequest");
+        expenseCalculatorServiceValidator.validateCalculatorSearchRequest(calculatorSearchRequest);
+
         RequestInfo requestInfo=calculatorSearchRequest.getRequestInfo();
+
         String tenantId=calculatorSearchRequest.getSearchCriteria().getTenantId();
 
         Map<String,BillMapper> billMappers=expenseCalculatorRepository.getBillMappers(calculatorSearchRequest);
-        List<Bill> bills=expenseCalculatorUtil.fetchBillsWithBillIds(requestInfo,tenantId,new ArrayList<>(billMappers.keySet()));
+
+        //set total count
+        calculatorSearchRequest.getPagination().setTotalCount(billMappers.size());
```

Comment:
don't do this, this should be the result from a separate count query ran for the same query with no pagination. remove the bills.size

Reply:
Change implemented in later commits.